src/blas/backends/generic/generic_level1.cxx

```diff
@@ -32,22 +32,38 @@ void dotu(sycl::queue& queue, std::int64_t n, sycl::buffer<std::complex<real_t>,
 }
 
 void iamax(sycl::queue& queue, std::int64_t n, sycl::buffer<real_t, 1>& x, std::int64_t incx,
-           sycl::buffer<std::int64_t, 1>& result) {
+           sycl::buffer<std::int64_t, 1>& result, oneapi::math::index_base base) {
     CALL_GENERIC_BLAS_FN(::blas::_iamax, queue, n, x, incx, result);
+    queue.submit([&](sycl::handler& cgh) {
```

Comment:
It's not ideal to submit another kernel for this. Ideally it should be done in the same kernel provided by https://github.com/uxlfoundation/generic-sycl-components/tree/main.
I understand this is more work. For now I would suggest to only submit that kernel if the `index_base` is `one` and add a TODO comment that this could be improved. Same comment for the other similar places in the generic backend.

Reply:
fixed in 10760de